.github/workflows/pre-commit.yml

```diff
@@ -0,0 +1,30 @@
+name: Pre-commit actions for Cocuhbase Plugin
+
+on: [pull_request]
+
+jobs:
+  pytest:
+    name: Test ${{ matrix.package }} on ${{ matrix.os }} using pytest
+    runs-on: ${{ matrix.os }}
+    strategy:
+      matrix:
+        python-version: [2.7]
+        os: [ubuntu-latest, macos-latest]
+
+    steps:
+    - name: Checkout ${{ matrix.package }} project
```

Comment:
nit: `matrix.package` is not defined. You could potentially add `package: [couchbase-plugin]` under `matrix` section.

Reply:
Thanks, @fdrozdowski. I have added below line:
package: [couchbase-plugin]